integration/disk_flush_helpers.go

```diff
@@ -85,7 +117,26 @@ func verifyForTime(
 	}
 	actual := make(generate.SeriesBlock, 0, len(expected))
 	for shard := range shards {
-		require.NoError(t, reader.Open(namespace, shard, timestamp))
+		rOpts := fs.ReaderOpenOptions{
+			Namespace:   namespace,
+			Shard:       shard,
+			BlockStart:  timestamp,
+			FilesetType: filesetType,
+		}
+
+		if filesetType == persist.FilesetSnapshotType {
```

Comment:
would it be better to push this logic into the reader ? 

you could have it so that by default it uses the latest version on disk, which can be overriden by passing a `SnapshotIndex`

Reply:
Discussed in person and decided this is a little too magical and leads to config confusion. I'd prefer not to implement it.